compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -6794,9 +6794,9 @@ private BType checkIndexAccessExpr(BLangIndexBasedAccess indexBasedAccessExpr) {
                     return symTable.semanticError;
                 }
             }
-
-            indexBasedAccessExpr.originalType = tableType.constraint;
-            actualType = tableType.constraint;
+            BType constraint = tableType.constraint;
+            indexBasedAccessExpr.originalType = constraint;
+            actualType = addNilForNillableAccessType(constraint);;
```

Comment:
@lasinicl, as discussed let's fix `indexBasedAccessExpr.originalType` for nested access similar to https://github.com/ballerina-platform/ballerina-lang/pull/28322/files#diff-f4fbb6e8ffab357b74e188781ad3f6e3235f8e48b64fba8dfb6ad72bf9ccaf93R6696.

Reply:
Fixed in 1d1917e

Reply:
Fixed in 1d1917e